tls/s2n_recv.c

```diff
@@ -179,7 +179,9 @@ ssize_t s2n_recv(struct s2n_connection * conn, void *buf, ssize_t size, s2n_bloc
                 GUARD(s2n_process_alert_fragment(conn));
                 GUARD(s2n_flush(conn, blocked));
             }
-
+            if (record_type == TLS_HANDSHAKE) {
```

Comment:
nit: else if or maybe a switch statement could work here. The record_type won't be both TLS_ALERT and TLS_HANDSHAKE.

Does this prevent KeyUpdate messages from being handled before the FINISHED message?